### ISSUE TITLE
Windows Agent Pool Nodes error

### ISSUE THREAD

**Issue by fredderf204** [Azure/acs-engine#4184]
**Is this a request for help?**:

Yes

**Is this an ISSUE or FEATURE REQUEST?** (choose one):

Issue

**What version of acs-engine?**:

v0.24.2

**Orchestrator and version (e.g. Kubernetes, DC/OS, Swarm)**

K8s 1.12.1

**What happened**:

When creating a cluster that has 1 Windows Agent Pool using VMSS, Agent Pool Nodes are not showing up when running the kubectl get nodes command

**What you expected to happen**:

VMSS Windows Agent Pool Nodes would show up when using the kubectl get nodes command

**How to reproduce it** (as minimally and precisely as possible):

Deploy a new cluster using K8s v 1.12.1, Windows Agent Pool, VMSS

**Anything else we need to know**:

The kubelet.err.log on the VMSS instances says

I1105 23:47:52.387646   11236 server.go:408] Version: v1.12.1
E1105 23:47:52.622640   11236 server.go:731] Kubelet needs to run as uid `0`. It is being run as -1
W1105 23:47:52.622640   11236 server.go:741] setting OOM scores is unsupported in this build
I1105 23:47:52.622640   11236 kubelet.go:304] Watching apiserver
W1105 23:47:52.624642   11236 iptables.go:155] Error checking iptables version, assuming version at least 1.4.11: executable file not found in %PATH%
I1105 23:47:52.626642   11236 client.go:104] Start docker client with request timeout=2m0s
E1105 23:47:52.626642   11236 kube_docker_client.go:91] failed to retrieve docker version: Error response from daemon: client version 1.38 is too new. Maximum supported API version is 1.37
W1105 23:47:52.627650   11236 kube_docker_client.go:92] Using empty version for docker client, this may sometimes cause compatibility issue.
F1105 23:47:52.627650   11236 server.go:262] failed to run Kubelet: failed to create kubelet: failed to get docker version: Error response from daemon: client version 1.38 is too new. Maximum supported API version is 1.37

**Comment by fredderf204** [comment]
Cluster Definition file below;
```
{
    "apiVersion": "vlabs",
    "properties": {
      "orchestratorProfile": {
        "orchestratorType": "Kubernetes",
        "orchestratorRelease": "1.12",
        "kubernetesConfig": {
          "useManagedIdentity": true
        }
      },
      "masterProfile": {
        "count": 1,
        "dnsPrefix": "mfvmsswin01",
        "vmSize": "Standard_D2_v2"
      },
      "agentPoolProfiles": [
        {
          "name": "agentpool1",
          "count": 1,
          "vmSize": "Standard_D2_v2",
          "availabilityProfile": "VirtualMachineScaleSets",
          "osType": "Windows",
          "storageProfile": "ManagedDisks"
        }
      ],
      "windowsProfile": {
        "adminUsername": "azureuser",
        "adminPassword": ""
      },
      "linuxProfile": {
        "adminUsername": "azureuser",
        "ssh": {
          "publicKeys": [
            {
            "keyData": ""
            }
          ]
        }
      }
    }
  }
```

**Comment by andyzhangx** [comment]
duplicated to https://github.com/kubernetes/kubernetes/issues/69996

**Comment by zhiweiv** [comment]
There is a workaround in #4118 , and it should be fixed in 0.25.

**Comment by fredderf204** [comment]
Confirming that workaround #4118 resolved this issue.
